website/guides/core-workflow.html.md

```diff
@@ -0,0 +1,335 @@
+---
+layout: "guides"
+page_title: "The Core Terraform Workflow - Guides"
+sidebar_current: "guides-core-workflow"
+description: |-
+  This guide provides an overview of the core Terraform workflow and how it
+  applies in individual, team, and organization contexts across Terraform open
+  source and Terraform Enterprise.
+---
+
+# The Core Terraform Workflow
+
+The core Terraform workflow has three steps:
+
+1. **Write** - Author infrastructure as code.
+2. **Plan** - Preview changes before applying.
+3. **Apply** - Provision reproducible infrastructure.
+
+This guide walks through how each of these three steps plays out in the context
+of working as an individual practitioner, how they evolve when a team is
+collaborating on infrastructure, and how Terraform Enterprise enables this
+workflow to run smoothly for entire organizations.
+
+## Working as an Individual Practitioner
+ 
+Let's first walk through how these parts fit together as an individual working
+on infrastructure as code.
+
+### Write
+
+You write Terraform configuration just like you write code: in your editor of
+choice. It's common practice to store your work in a version controlled
+repository even when you're just operating as an individual.
+
+```sh
+# Create repository
+$ git init my-infra && cd my-infra
+
+Initialized empty Git repository in /.../my-infra/.git/
+
+# Write initial config
+$ vim main.tf
+
+# Initialize Terraform
+$ terraform init
+
+Initializing provider plugins...
+# ...
+Terraform has been successfully initialized!
+```
+
+As you make progress on authoring your config, repeatedly running plans can help
+flush out syntax errors and ensure that your config is coming together as you
+expect.
+
+```sh
+# Make edits to config
+$ vim main.tf
+
+# Review plan
+$ terraform plan
+
+# Make additional edits, and repeat
+$ vim main.tf
+```
+
+This parallels working on application code as an individual, where a tight
+feedback loop between editing code and running test commands is useful.
+
+### Plan
+
+When the feedback loop of the Write step has yielded a change that looks good,
+it's time to commit your work and review the final plan.
+
+```sh
+$ git add main.tf
+$ git commit -m 'Managing infrastructure as code!'
+
+[master (root-commit) f735520] Managing infrastructure as code!
+ 1 file changed, 1 insertion(+)
+```
+
+Because `terraform apply` will display a plan for confirmation before
+proceeding to change any infrastructure, that's the command you run for final
+review.
+
+```sh
+$ terraform apply
```

Comment:
I get what's going on here after reading it a couple times, but it initially confused me to see `terraform apply` under the "Plan" heading.

I'm honestly not really sure what other way this could be written, since moving this step into the next section would of course leave no Terraform commands at all in this section, but maybe there's some way to make it more explicit here that this section actually ends _while `teraform apply` is still running_, and that the following section therefore picks up immediately within the same command.

Reply:
Yeah I'll think on this and follow up next week with fresh eyes.